pkg/controller/appdefinition/parse.go

```diff
@@ -12,11 +14,24 @@ func ParseAppImage(req router.Request, resp router.Response) error {
 	appInstance := req.Object.(*v1.AppInstance)
 	status := condition.Setter(appInstance, resp, v1.AppInstanceConditionParsed)
 	appImage := appInstance.Status.AppImage
+	containers := appImage.ImageData.Containers
 
 	if appImage.Acornfile == "" {
 		return nil
 	}
 
+	// TODO: move this to a fucntion
+	for _, p := range appInstance.Spec.Publish {
+		if p.TargetServiceName != "" && len(containers) > 0 {
+			if _, ok := containers[p.TargetServiceName]; ok {
+				continue
+			} else {
+				status.Error(fmt.Errorf("invalid container name in pubish flag - received [%s]", p.TargetServiceName))
+				return nil
+			}
```

Comment:
This if-else block can be simplified. There is no need for an else statement.
```suggestion
				if _, ok := containers[p.TargetServiceName]; !ok {
				          status.Error(fmt.Errorf("invalid container name in pubish flag - received [%s]", p.TargetServiceName))
				         return nil
			       } 
			       continue
```